includes/class-newspack-popups-model.php

```diff
@@ -1186,6 +1189,27 @@ function ( $evts ) use ( $popup, $body, $element_id ) {
 
 		$animation_id = 'a_' . $element_id;
 
+		// If the dismiss action is by form submission, dismiss the prompt when the form is submitted.
+		if ( 'form_submission' === $dismiss_action ) {
+			$site_is_amp         = Newspack_Popups_Inserter::is_amp() && ! Newspack_Popups_Inserter::is_amp_plus();
+			$newspack_form_class = apply_filters( 'newspack_campaigns_form_class', 'newspack-subscribe-form' );
+			$has_form            = preg_match( '/<form\s|mc4wp-form|\[gravityforms\s|' . $newspack_form_class . '/', $body ) !== 0;
```

Comment:
Both MC4WP and GF plugins' output an HTML form. Isn't `|mc4wp-form|\[gravityforms\s` in the RegExp redundant?